packages/legacy-client/src/client/baseTypes/ActionType.ts

```diff
@@ -179,3 +201,52 @@ export const ActionResponse = {
     return { validation } as ActionResponse;
   },
 };
+
+export const BulkActionResponse = {
+  of: <
+    TAddedObjects extends OntologyObject<string, NonNullable<ParameterValue>>,
+    TModifiedObjects extends OntologyObject<
+      string,
+      NonNullable<ParameterValue>
+    >,
+  >(
+    client: ClientContext<OntologyDefinition<any>>,
+    response: BatchApplyActionResponseV2,
+  ): BulkActionResponse<Edits<TAddedObjects, TModifiedObjects> | undefined> => {
+    if (response.edits?.type === "edits") {
+      const added: any[] = [];
+      const modified: any[] = [];
+      for (const edit of response.edits.edits) {
```

Comment:
This seems like a copy/paste of the above.  Maybe extract function for getting the `edits` shape (`{type:"edits", added,modified}`)?